packages/components/pie-button/src/index.ts

```diff
@@ -55,6 +55,22 @@ export class PieButton extends LitElement implements ButtonProps {
     updated (changedProperties: PropertyValues<this>): void {
         super.updated(changedProperties);
 
+        // Making sure that icon passed in a slot has the correct icon size
```

Comment:
Testing locally, I managed to get this to work adding it to the render method and onslotchange. Unfortunately I cannot test this in SSR as we have a bug to fix first that blocks testing but we can pick this up in the new year

Reply:
I will move this change to another PR to fix current bug using css vars

Reply:
Here is a draft PR for this: https://github.com/justeattakeaway/pie/pull/1128